### PR TITLE
Support private modules

### PR DESCRIPTION
**Pull request by burke**

### CODE REVIEW

File: lib.nix [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 let
-  modules = [ runix-config packages ] ++ (import ./modules/module-list.nix);
+  # overlay is a function taking a set of modules and returning the final set of modules.
+  # Typically it will look like
+  #     modules: modules ++ (import ./modules/module-list.nix)
+  # ...but it can do whatever it wants. If none was provided, we use Identity.
+  modules = let
+    res = builtins.tryEval (import runix-overlay);
+    fn = if res.success then res.value else (f: f);
+  in fn ([ runix-config packages ] ++ (import ./modules/module-list.nix));
```
Comment (burke) [inline lib.nix]: There are a lot of conceivable ways to organize this. Giving it a function over the list of module paths seemed like a good flexibility / complexity tradeoff at the moment but I'm very open to discussion on this point 🤷‍♂

File: README.md [hunk 2]
```diff
@@ -82,6 +82,28 @@ For Shopify developers, there is an experimental branch of `dev` ("ifd") with
...
+## Private Module Overlays
+
+Runix supports private module overlays.
+
+```
+export RUNIX_OVERLAY=/path/to/my/module-overlay.nix
+```
+
+This file should evaluate to a function which takes as an input the default
+module list generated by Runix, and return your amended module set.
```
Comment (klautcomputing) [inline README.md]:
```suggestion
module list generated by Runix, and returns your amended module set.
```

File: README.md [hunk 3]
```diff
@@ -82,6 +82,28 @@ For Shopify developers, there is an experimental branch of `dev` ("ifd") with
...
+
+In general, you probably want something like this:
+
+``` nix
+modules: modules ++ (import ./modules/modules-list.nix)
+```
+
+...and then in the `modules` directory and that `modules-modules-list.nix` file,
```
Comment (klautcomputing) [inline README.md]: I think this could be a little clearer if worded differently.

File: lib.nix (hunk 1 above)
Comment (honkfestival) [reply lib.nix]: It's as good a start as any and we'll need to get people using Runix in anger before 'better' becomes apparent.

**Pull request merged**